src/common/multihash_isal.c

```diff
@@ -317,8 +318,8 @@ struct hash_ft sha1_algo = {
 
 /** SHA256 */
 struct sha256_ctx {
-	struct mh_sha256_ctx	s2_ctx;
-	bool			s2_updated;
+	struct isal_mh_sha256_ctx s2_ctx;
+	bool                      s2_updated;
 };
 
 static int
```

Comment:
Can we also fix the `sha256_init` function to eliminate potential memory leak in case of the failure of `isal_mh_sha256_init`?
 ````suggestion
static int
sha256_init(void **daos_mhash_ctx)
{
	struct sha256_ctx	*ctx;
	int			 rc;

	D_ALLOC_PTR(ctx);
	if (ctx == NULL)
		return -DER_NOMEM;

	rc = sha256_reset(&ctx->s2_ctx);
	if (rc != 0) {
		sha256_destroy(ctx);
		return rc;
	}
	*daos_mhash_ctx = ctx;
	return 0;
}